mitodl/mit-open
# do not allow None in levels/languages

### What are the relevant tickets?
#432 

### Description (What does it do?)
Removes null values from `levels` and `languages` arrays coming from OLL.

### How can this be tested?
1. If you don't have them already, get the values of 
    ```
    EDX_API_CLIENT_ID
    EDX_API_CLIENT_SECRET
    EDX_API_ACCESS_TOKEN_URL
    EDX_API_URL
    EDX_BASE_URL
    EDX_ALT_URL
    OLL_API_CLIENT_ID
    OLL_API_CLIENT_SECRET
    OLL_API_ACCESS_TOKEN_URL
    OLL_API_URL
    OLL_BASE_URL
    OLL_ALT_URL
    ```
    from RC.
2. On `main`, run `backpopulate_oll_data` and `backpopulate_mit_edx_data`.
3. In a django shell,
    ```
    from learning_resources.models import LearningResourceRun
    LearningResourceRun.objects.filter(level=[None])
    ```
    there should be some and visit http://localhost:8063/api/v1/courses/?platform=oll ... you should see runs have nulls in `language` and `level`.
4. Restart containers on this branch.
5. Re-run the query from (3). There should no longer be any; visit  http://localhost:8063/api/v1/courses/?platform=oll  again; level and language arrays should not have nulls. 
6. Run `backpopulate_oll_data --delete` and then `backpopulate_oll_data` to re-import the data afresh. Should not have nulls.
